### PR TITLE
fix typo in AssertionErrorMessagesAggregrator

### PR DESCRIPTION
**Pull request by ahmadmoawad**
#### Check List:

* Fix typo in `AssertionErrorMessagesAggregrator` class and `aggregrateErrorMessages()` method
* Unit tests :  NO
* Javadoc with a code example (on API only) : NO

### CODE REVIEW

File: src/main/java/org/assertj/core/error/AssertionErrorMessagesAggregator.java [hunk 1]
```diff
@@ -14,9 +14,9 @@
 
 import java.util.List;
 
-public class AssertionErrorMessagesAggregrator {
+public class AssertionErrorMessagesAggregator {
```
Comment (scordio) [inline src/main/java/org/assertj/core/error/AssertionErrorMessagesAggregator.java:17]: Not being in the internal package, this class might be used by other projects/libraries and the renaming would be a breaking change. We should probably deprecate this one and create a new class with the correct namings.

@joel-costigliola do you agree?

Comment (joel-costigliola) [reply src/main/java/org/assertj/core/error/AssertionErrorMessagesAggregator.java:17]: I agree!

**Comment by ahmadmoawad** [comment]
@scordio ,

Should I mark `AssertionErrorMessagesAggregrator` class as deprecated and put a message referring to the new one `AssertionErrorMessagesAggregator`?

**Comment by scordio** [comment]
@ahmadmoawad yes, thank you. Summarizing, I would suggest the following changes:

- Create a new `AssertionErrorMessagesAggregator` with `aggregateErrorMessages()`
- Let `SoftAssertionError` and `MultipleAssertionsError` use the new class
- Deprecate `AssertionErrorMessagesAggregrator` with javadoc referring to the new class

Please let us know if you have any questions.

**Comment by ahmadmoawad** [comment]
Okay @scordio , I will start implementing them

**Comment by ahmadmoawad** [comment]
@scordio , Could you please review the changes

**Comment by scordio** [comment]
Merged, thanks @ahmadmoawad!

File: src/main/java/org/assertj/core/error/AssertionErrorMessagesAggregrator.java [hunk 2]
```diff
@@ -14,6 +14,10 @@
 
 import java.util.List;
 
+/**
+ * @deprecated use {@link AssertionErrorMessagesAggregator} instead
+ */
+@Deprecated
 public class AssertionErrorMessagesAggregrator {
 
   public static String aggregrateErrorMessages(List<String> errors) {
```
Comment (pzygielo) [inline src/main/java/org/assertj/core/error/AssertionErrorMessagesAggregrator.java:23]: Perhaps method from replacement class could be called here to stay DRY, and have no other code in this class?

Comment (scordio) [reply src/main/java/org/assertj/core/error/AssertionErrorMessagesAggregrator.java:23]: Done in 45eb16da4a2290a0d3f9c77f7806372507705af6, thanks @pzygielo!

**Pull request merged**